packages/react-devtools-extensions/src/generateHookMap.js

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import traverse, {Node, NodePath} from '@babel/traverse';
+import {string} from 'yargs';
+import {getHookNamesFromAST} from './astUtils';
+import {encode, decode} from 'sourcemap-codec';
```

Comment:
We're relying on the fact that we need to encode/decode tuples of a size that `sourcemap-codec` happens to support (because it assumes the thing being encoded/decoded is a standard source map entry). I guess we could keep doing that if it works, but note that it would only take adding one tuple element to our data (5 --> 6) or removing two (5 --> 3) to break this assumption. Long term I'd recommend writing our own codec.

Reply:
yep that's correct `encode`/`decode` require using tuples of the size of the standard source map entry  (i'll make sure to add a comment here if we opt for continuing to use `encode`). 

i went with this approach to try to get something working quickly, but yeah happy to add our own encoding implementation. do you have any thoughts on whether we should go ahead and do this straight away for v1? cc @bvaughn

Reply:
I don't have a strong opinion here.

I wonder if we could add something small to signify which version of the encoding a source map is using (to later help us if we change)? This could maybe be an extra field if we went with the more generic `x_react` attribute?